test/powershell/Language/Scripting/OutErrorVariable.Tests.ps1

```diff
@@ -0,0 +1,390 @@
+        Validate-Result $foo_err1 '$foo_err1' 1 @("foo-message")
+        Validate-Result $foo_err2 '$foo_err2' 1 @("foo-message")
+        Validate-Result $foo_err3 '$foo_err3' 1 @("foo-message")
+    }
+
+    Context 'Error variable in multi-command pipeline (with native cmdlet)' {
+
+        (get-foo -ev foo_err | get-item -ev get_item_err ) 2>&1 > $null
+
+        Validate-Result $foo_err '$foo_err' 1 @("foo-error")
+
+        It '$get_item_err.count' { $get_item_err.count | Should Be 1 }
+        It '$get_item_err[0].exception' { $get_item_err[0].exception.GetType() | Should Be 'System.Management.Automation.ItemNotFoundException' }
+    }
+
+    Context 'Multi-command pipeline with nested commands' {       
+
+        function get-bar3 
+        {
+            [CmdletBinding()]
+            param([Parameter(ValueFromPipeline = $true)][string] $i)
+
+            write-error  "bar-error"
+            write-output "bar-output"
+            get-foo
+        }
+
+        (get-foo -errorVariable foo_err | get-bar3 -errorVariable bar_err) 2>&1 > $null        
+
+        Validate-Result $foo_err '$foo_err' 1 @("foo-error")
+
+        Validate-Result $bar_err '$bar_err' 2 @("bar-error", "foo-error")
+}
+
+    Context 'multi-command pipeline with nested commands' { 
+
+        function get-foo6
+        {
+            [CmdletBinding()]
+            param([Parameter(ValueFromPipeline = $true)][string] $i)
+
+            write-error  "foo-error"
+            write-output $i
+        }
+
+        function get-bar4
+        {
+            [CmdletBinding()]
+            param([Parameter(ValueFromPipeline = $true)][string] $i)
+
+            write-error  "bar-error"
+            get-foo6 "foo-output" -errorVariable global:foo_err1 | get-foo6 -errorVariable global:foo_err2 
+        }
+
+        get-bar4 -errorVariable global:bar_err 2>&1 > $null
+
+        Validate-Result $global:foo_err1 '$global:foo_err1' 1 @("foo-error")
+        Validate-Result $global:foo_err2 '$global:foo_err2' 1 @("foo-error")
+
+        Validate-Result $global:bar_err '$global:bar_err' 2 @("bar-error", "foo-error")        
+    }
+
+    Context 'Nested output variables' {
+        BeforeAll {
+            
+
+            function get-foo7
+            {
+                [CmdletBinding()]
+                param([Parameter(ValueFromPipeline = $true)][string] $output)
+
+                $output
+                write-error "foo-error"
+            }
+
+            function get-bar5
+            {
+                [CmdletBinding()]
+                param()
+
+                "bar-output"
+                write-error  "bar-error"
+                get-foo7 "foo-output" -ev global:foo_err1 -ov global:foo_out1 | get-foo7 -ev global:foo_err2 -ov global:foo_out2
+                get-foo7 "foo-output" -ev global:foo_err3 -ov global:foo_out3 | get-foo7 -ev global:foo_err4 -ov global:foo_out4
+            }
+        }
+
+        get-bar5 -ev global:bar_err -ov global:bar_out 2>&1 > $null
```

Comment:
i updated some of the tests via test case. some other tests validate different number of variables.  Changing to test case does not help to reduce the code. 

---
In reply to: [77278104](https://github.com/Powershell/powershell/pull/2160#discussion_r77278104) [](ancestors = 77278104)